selfhost/typechecker.jakt

```diff
@@ -2811,9 +2820,14 @@ struct Typechecker {
 
         if not structure_linkage is External and not return_type_id.equals(VOID_TYPE_ID) and not block.definitely_returns {
             // FIXME: Use better span
-            .error("Control reaches end of non-void function", func.name_span)
+            if return_type_id.equals(never_type_id()) and not block.never_returns {
+                .error("Control reaches end of never-returning function", func.name_span)
```

Comment:
I suggest a different message, like "never-returning function should not return"

Reply:
that message would not make sense for a void returning function

Reply:
I'm talking about the message for a function that never returns, I thought it was clear by the context I'm giving.